autoload/bazel/lib.vim

```diff
@@ -0,0 +1,52 @@
+""
+" @section Introduction, intro
+" @library
+" Implements common functionality used by bazel-related plugins.  For more
+" details about Build file structure/terminology, see:
+" https://bazel.build/versions/master/docs/build-ref.html
+
+
+""
+" @private
+" Returns the root path for Bazel workspace. {path} is the absolute path.
+"
+" I.e., if you are in directory: /Users/$USER/myproject/foo/bar/biff
+" And there is a WORKSPACE file at: /Users/$USER/myproject/WORKSPACE
+" Return /home/$USER/myproject. If no workspace can be found, return ''
+function! bazel#lib#GetBazelRootPath(path) abort
+  let l:path = s:AbsoluteDirectoryPath(a:path)
+  if empty(l:path)
+    return l:path
+  endif
+  let l:components = maktaba#path#Split(path)
+  let l:working = ''
+  let l:workspace_path = ''
+  for l:comp in l:components
```

Comment:
Would it work to use `findfile()` instead of a custom traversal loop in vimscript?